metrics-operator/converter/slo_converter.go

```diff
@@ -136,24 +139,21 @@ func removePercentage(str string) (int, error) {
 }
 
 // creates and sets up the target struct from objective
-// TODO refactor this function in a follow-up + weight distribution
 // nolint:gocognit,gocyclo
 func setupTarget(o *Objective) (*metricsapi.Target, error) {
 	target := &metricsapi.Target{}
-	// remove criteria, which contain % in their operators
-	o = cleanupObjective(o)
 	// skip objective target conversion if it has criteria combined with logical OR -> not supported
```

Comment:
I think a great addition to simplify the understanding of the code is a `doc.go` file with some ASCII ART graphs that depict each different case how is converted. 
You can use https://asciiflow.com/#/

Reply:
agree, I want to follow up on this PR to create a docs description with all the cases which are covered and which are not supported.